frontends/systemverilog/uhdm_ast.cc

```diff
@@ -2839,6 +2846,43 @@ void UhdmAst::process_array_var()
         } else if (vpi_get(vpiType, reg_h) == vpiIntVar) {
             packed_ranges.push_back(make_range(31, 0));
             visit_default_expr(reg_h);
+        } else if (vpi_get(vpiType, reg_h) == vpiPackedArrayVar) {
+            vpiHandle itr2 = vpi_iterate(vpi_get(vpiType, reg_h) == vpiArrayVar ? vpiReg : vpiElement, reg_h);
+            while (vpiHandle reg_h2 = vpi_scan(itr2)) {
+                if (vpi_get(vpiType, reg_h2) == vpiStructVar || vpi_get(vpiType, reg_h2) == vpiEnumVar) {
+                    visit_one_to_one({vpiTypespec}, reg_h2, [&](AST::AstNode *node) {
+                        if (node->str.empty()) {
+                            // anonymous typespec, move the children to variable
+                            current_node->type = node->type;
+                            current_node->children = std::move(node->children);
+                        } else {
+                            auto wiretype_node = new AST::AstNode(AST::AST_WIRETYPE);
+                            wiretype_node->str = node->str;
+                            current_node->children.push_back(wiretype_node);
+                            current_node->is_custom_type = true;
+                        }
+                        delete node;
+                    });
+                } else if (vpi_get(vpiType, reg_h2) == vpiLogicVar) {
+                    current_node->is_logic = true;
+                    visit_one_to_one({vpiTypespec}, reg_h2, [&](AST::AstNode *node) {
+                        if (node->str.empty()) {
+                            // anonymous typespec, move the children to variable
+                            current_node->type = node->type;
+                            current_node->children = std::move(node->children);
+                        } else {
+                            auto wiretype_node = new AST::AstNode(AST::AST_WIRETYPE);
+                            wiretype_node->str = node->str;
+                            current_node->children.push_back(wiretype_node);
+                            current_node->is_custom_type = true;
+                        }
+                        delete node;
```

Comment:
```suggestion
                 auto vpi_obj = vpi_get(vpiType, reg_h2);
                 if (vpi_obj == vpiStructVar || vpi_obj == vpiEnumVar || vpi_obj == vpiLogicVar) {
                     if (vpi_obj == vpiLogicVar)
                         current_node->is_logic = true;
                     visit_one_to_one({vpiTypespec}, reg_h2, [&](AST::AstNode *node) {
                         if (node->str.empty()) {
                             // anonymous typespec, move the children to variable
                             current_node->type = node->type;
                             current_node->children = std::move(node->children);
                         } else {
                             auto wiretype_node = new AST::AstNode(AST::AST_WIRETYPE);
                             wiretype_node->str = node->str;
                             current_node->children.push_back(wiretype_node);
                             current_node->is_custom_type = true;
                         }
                         delete node;
```

With this we can avoid code duplication + `vpi_get` will be called only once (unless I'm missing something)